### PR TITLE
Added Schedular For Subscriptions

### PR DESCRIPTION
**Pull request by Aditya3034**
### Functionality:
Added schedular to check Subscription data.
### Solution:
Implemented mechanism to add Schedular which checks the subscriptions data periodically. 
### Risk level:
- [ ] high 
- [X] medium
- [ ] low

### How to test:
1.Open Emulator
2.Use the schedule button in my-dashboard component by Uncommenting it.
3.Check the Firebase logs.
4.You can see the scheduledCheckSub Api executing with no error.
5.You can check Subscription Status changing to Inactive after graceNotification <=0.

### CODE REVIEW

File: functions/model/subscriptions/tark/checkSubscription.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+      const graceNotifications = subData[0].GraceNotifications;
+      const subscriptionId = subData[0].SubscriptionId;
+
+      if (subData[0].SubscriptionType == "Basic") {
+        subData[0] = basicSubscription;
+      } else if (subData[0].SubscriptionType == "Standard") {
+        subData[0] = standardSubscription;
+      } else {
+        console.error("Error getting Subscription Plan!");
+      }
```
Comment (sanjaykrishna1203) [inline functions/model/subscriptions/tark/checkSubscription.js:24]: you should not assign the comparison data to the subData

File: functions/model/subscriptions/tark/checkSubscription.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+      const subCheck = [
+        TotalNumberOfTeams <= noOfTeams,
+        TotalNumberOfMembers <= noOfMembers,
+        CurrentDate < expiresOn
+      ]
```
Comment (sanjaykrishna1203) [inline functions/model/subscriptions/tark/checkSubscription.js:30]: This comparison must be done with the subData and the reference data from application

File: functions/model/subscriptions/tark/checkSubscription.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
+const { getApplicationData, currentDate, standardSubscription, basicSubscription } = require("../../application/lib");
+const { getSubscriptions, updateSubscription } = require("../lib");
+
+
+exports.checkSubscription = function (orgAppKey, subscriptionId) {
+  const p = getApplicationData().then((appData) => {
+    const TotalNumberOfTeams = appData.TotalNumberOfTeams;
+    const TotalNumberOfMembers = appData.TotalNumberOfMembers;
```
Comment (sanjaykrishna1203) [inline functions/model/subscriptions/tark/checkSubscription.js:8]: There is no need for these data here

**Review by sanjaykrishna1203 — CHANGES_REQUESTED** [review]
Please do the required changes

File: functions/model/organization/tark/createMember.js [hunk 4]
```diff
@@ -35,6 +38,19 @@
      const DateOfOnboarding = yyyy + "-" + mm + "-" + dd;
      const DateOfExit = "xxxx-xx-xx";
 
+
+     getApplicationData().then((data) => {
```
Comment (viv9k) [inline functions/model/organization/tark/createMember.js:42]: This is should be for org not for app

**Review by viv9k — CHANGES_REQUESTED** [review]
Need to improve and show a demo after this.

File: functions/model/organization/tark/createMember.js (hunk 4 above)
Comment (Aditya3034) [reply functions/model/organization/tark/createMember.js]: did the changes sir.

File: functions/model/subscriptions/tark/checkSubscription.js [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+      const graceNotifications = subData[0].GraceNotifications;
+      const subscriptionId = subData[0].SubscriptionId;
+
+      if (subData[0].SubscriptionType == "Basic") {
+        subData[0] = basicSubscription;
+      } else if (subData[0].SubscriptionType == "Standard") {
+        subData[0] = standardSubscription;
+      } else {
+        console.error("Error getting subscription plan!");
+      }
```
Comment (sanjaykrishna1203) [inline functions/model/subscriptions/tark/checkSubscription.js:37]: This is not needed

**Review by sanjaykrishna1203 — CHANGES_REQUESTED** [review]

File: functions/model/subscriptions/tark/checkSubscription.js (hunk 5 above)
Comment (Aditya3034) [reply functions/model/subscriptions/tark/checkSubscription.js:37]: okay bhaiya

**Review by viv9k — APPROVED** [review]
Approved

**Pull request merged**